api/views/event.py

```diff
@@ -119,3 +119,30 @@ def import_event(self, request):
             self.get_serializer(cloned_event).data,
             status=status.HTTP_201_CREATED,
         )
+
+    @action(detail=True, method=["get"], url_path="event-leader-boards")
+    def event_leader_boards(self, request, pk):
```

Comment:
name should be `leader_board` when it's in the course it's the course leader board and when it's in an event it's the event leader board.

/api/event/1/leader-board
/api/course/1/leader-board

Reply:
I'm trying to get all leaderboards for all events of a course.
That way we don't have to call the API every time a student switches to see another challenge. Is this a bad idea?

Reply:
In that case, it should be in the course and the url will be `/api/course/1/event-leader-boards`.
As the leaderboard calculation is significant. it might still be better to request them 1 by 1 as we are not showing them at the same time.